cmd/rook/ceph/osd.go

```diff
@@ -214,9 +214,11 @@ func prepareOSD(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	backgroundCtx := ctx.Background()
+
 	context := createContext()
 	commonOSDInit(provisionCmd)
-	crushLocation, topologyAffinity, err := getLocation(context.Clientset)
+	crushLocation, topologyAffinity, err := getLocation(backgroundCtx, context.Clientset)
```

Comment:
Why not use `cmd.Context()` like done in the other PR?

Reply:
I've updated the PR, I updated `prepareOSD` and `removeOSDs`.